pkg/snapstore/s3_snapstore.go

```diff
@@ -73,11 +76,14 @@ func newS3FromSessionOpt(bucket, prefix, tempDir string, maxParallelChunkUploads
 }
 
 func getSessionOptions(prefixString string) session.Options {
+	s3path, _ := strconv.ParseBool(os.Getenv(awsForcePathStyle))
```

Comment:
```suggestion
        var s3path *bool
	if val, err := strconv.ParseBool(os.Getenv(awsForcePathStyle)); err != nil {
	  s3pathPtr = &val
	}
```
To keep `S3ForcePathStyle` nil as it used to be before + we have a slightly improved error handling.

Reply:
My code looked somewhat like that, initially! 😅 However, I felt more confident about changing it to the way that it looks now after I looked at the `ParseBool()` source code (https://cs.opensource.google/go/go/+/refs/tags/go1.16:src/strconv/atob.go;l=10).

If the env var doesn't have any value (which is the default configuration when connecting to Amazon S3), then the function will return `false` and an error. The `false` is what I _really_ want from that, and the error doesn't really mean anything to me, since I expected it in that situation.

I think that if I were trying to give feedback to the user about the config being wrong, the error would be somewhat useful, but I don't think that's necessary in this case, and would require a much larger refactor than what's shown here.

Reply:
If you'd like, I can leave a comment on that line, explaining that intent to make it less confusing in the future.

Reply:
TBH, I don't see any benefits having a one liner + any comments.
With the suggestion above we:
- Pass `nil` like we do today if the env var is not set (-> no regression guarantee).
- We handle the error to some extent by keeping the `nil` value.
- One should not first look at the `ParseBool` implementation to figure out the first returned value but instead handle the error if one is returned as this is a very common pattern (see [Effective Go](https://go.dev/doc/effective_go) -> Multiple return values)

Reply:
Sounds good to me, I tested it locally and it worked. 👍